pinot-common/src/main/java/org/apache/pinot/common/version/PinotVersion.java

```diff
@@ -45,6 +45,16 @@ private PinotVersion() {
     // private constructor for utility class
   }
 
+  /**
+   * Returns a sanitized version string with all dots replaced with underscores, which is necessary
+   * for Prometheus to be able to properly handle the version.
+   *
+   * @return the version in a way that can be indexed by prometheus
+   */
+  public static String getVersionMetricName() {
+    return VERSION.replace('.', '_').replace('-', '_');
```

Comment:
Make a constant for `VERSION_METRIC_NAME` to avoid string manipulation per invocation